src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -104,9 +110,9 @@ def _get_docker_status_and_version(ignore_errors, yes):
                 logger.warning("Skipping pull check.")
                 return
 
-        output, warning, stderr = _subprocess_communicate([docker_command, "pull", IMAGE])
+        output, warning, stderr, succeeded = _subprocess_communicate([docker_command, "pull", IMAGE])
 
-        if stderr:
+        if not succeeded:
```

Comment:
Question: when the command is succeeded, do we still want to check the output message (line 124-126)?

Reply:
My concern is that what if the output message changes in the future?

Reply:
Nice catch. I agree that the success check might break if the output message changes in the future. Therefore, I’m considering removing this check. I'm thinking of removing this check and changes to:

```
        if not succeeded:
            _handle_error(DOCKER_PULL_ERROR.append_error_message(stderr), ignore_errors)
        else:
            if DOCKER_PULL_WRONG_PLATFORM in stderr:
                print_pass("Docker pull of '{}'".format(IMAGE))
                logger.warning("Image '%s' can be pulled but cannot be used on this platform", IMAGE)
                return
            if warning:
                logger.warning(warning)
            print_pass("Docker pull of '{}'".format(IMAGE))
```

Do you have any suggestions? Thanks

Reply:
I think this works. If the platform is unmatched, would the exit code be zero or non-zero?

Reply:
Discussed offline. For the DOCKER_PULL_WRONG_PLATFORM error, the exit code is 1, I've moved this logic under "not succeeded" case.